components/Tag/Tag.jsx

```diff
@@ -38,37 +20,50 @@ class Tag extends Component {
 
   constructor() {
     super();
+
     this.state = {
-      isHover: false,
-      isRemoveHover: false,
-    };
-    this.setHover = this.setHover.bind(this);
-    this.setRemoveHover = this.setRemoveHover.bind(this);
+      isHovering:false,
+      isHoveringOverX:false
+    }
+
+    this.handleRemove = this.handleRemove.bind(this);
+    this.getXColor = this.getXColor.bind(this);
   }
 
-  setHover(value) {
-    return () => this.setState({ isHover: value });
+  handleRemove (event) {
+
+    console.log('click')
+    if(this.props.onRemove) {
+      this.props.onRemove();
+    }
   }
 
-  setRemoveHover(value) {
-    return () => this.setState({ isRemoveHover: value });
+  getXColor (isHovering, isHoveringOverX) {
+    if(isHoveringOverX) {
```

Comment:
We could split this up into 2 `if` statements and a default return so that we don't have to use `else if`.

i.e.;
```
if (isHoveringOverX) return null;
if (isHovering) return 'white';
return 'gray';
```